AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/script_canvas_tools_qt.py

```diff
@@ -0,0 +1,448 @@
+    """
+    function used for expanding qt container rows with expandable children
+
+    param self: The script object calling this function
+    param object_name: qt object name as a string
+
+    returns: none
+    """
+    children = self.asset_editor_row_container.findChildren(QtWidgets.QFrame, object_name)
+    for child in children:
+        check_box = child.findChild(QtWidgets.QCheckBox)
+        if check_box and not check_box.isChecked():
+            QtTest.QTest.mouseClick(check_box, QtCore.Qt.LeftButton, QtCore.Qt.NoModifier)
+
+
+def open_node_palette(self):
+    """
+    function for checking if node palette is on and if not turn it on
+
+    param self: the script calling this function
+
+    returns none
+    """
+    if self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT) is None:
+        action = pyside_utils.find_child_by_pattern(self.sc_editor, {"text": NODE_PALETTE_UI, "type": QtWidgets.QAction})
+        action.trigger()
+
+
+def open_asset_editor():
+    """
+    function for opening the asset editor UI
+
+    returns true/false result of helper function's attempt
+    """
+    general.open_pane(ASSET_EDITOR_UI)
+    result = helper.wait_for_condition(lambda: general.is_pane_visible(ASSET_EDITOR_UI), WAIT_TIME_3)
+    return result
+
+
+def canvas_node_palette_search(self, node_name, number_of_retries):
+    """
+    function for searching the script canvas node palette for user defined nodes. function takes a number of retries as
+    an argument in case editor/script canvas lags during test.
+
+    param self: The script calling this function
+    param node_name: the name of the node being searched for
+    param number_of_retries: the number of times to search (click on the search button)
+
+    returns: boolean value of the search attempt
+    """
+    self.node_tree_search_box.setText(node_name)
+    helper.wait_for_condition(lambda: self.node_tree_search_box.text() == node_name, WAIT_TIME_3)
+    # Try clicking ENTER in search box multiple times
+    found_node = False
+    for _ in range(number_of_retries):
+        QtTest.QTest.keyClick(self.node_tree_search_box, QtCore.Qt.Key_Enter, QtCore.Qt.NoModifier)
+        found_node = helper.wait_for_condition(
+            lambda: pyside_utils.find_child_by_pattern(self.node_tree_view, {"text": node_name}) is not None, WAIT_TIME_3)
+        if found_node is True:
+            break
+    return found_node
+
+def get_node_palette_node_tree_qt_object (self):
+    """
+    function for retrieving the tree view qt object for the node palette
+
+    params self: the script calling this function
+
+    returns: a tree view qt object
+    """
+    node_palette_widget = self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT)
+    node_palette_node_tree = node_palette_widget.findChild(QtWidgets.QTreeView, TREE_VIEW_QT)
+    return node_palette_node_tree
+
+
+def get_node_palette_category_qt_object(self, category_name):
```

Comment:
Can this be refactorted into SC Editor or its own QtPy class?

Reply:
Yep! I plan on putting all the sc tools into their own classes and adding references to them on the SC editor.